Modules/_localemodule.c

```diff
@@ -269,7 +274,12 @@ PyLocale_strxfrm(PyObject* self, PyObject* args)
             goto exit;
         }
         buf = new_buf;
+        errno = 0;
         n2 = wcsxfrm(buf, s, n2+1);
+        if (errno) {
+            PyErr_SetFromErrno(PyExc_OSError);
+            goto exit;
+        }
     }
     result = PyUnicode_FromWideChar(buf, n2);
 exit:
```

Comment:
While you modify this function, can you please remove useless "if (buf)" and "if (s)" below? PyMem_Free(NULL) is well defined: it does nothing.